src/main/java/com/example/gradecalculator/service/UserService.java

```diff
@@ -3,7 +3,7 @@
 import com.example.gradecalculator.entities.User;
 import com.example.gradecalculator.entities.UserType;
 import com.example.gradecalculator.mapper.UserRegistrationMapper;
-import com.example.gradecalculator.model.UserDetailsImpl;
+import com.example.gradecalculator.mapper.UserDetailsImpl;
```

Comment:
this seems wrong. There should be no more 'UserDetailsImpl' and mapper is the wrong directory for service anyway